src/controllers/sdex/index.ts

```diff
@@ -72,4 +79,23 @@ router.get(
   }),
 );
 
+router.get(
+  '/swaps',
+  asyncRoute(async (req, res) => {
+    const { user, chainId, cursor, limit } = validate(swapHistoryQuerySchema, req.query);
+
+    const cacheKey = `sdex/swaps/${chainId}/${user}/${cursor || 0}/${limit}`;
+
+    return maybeCacheResponse(
+      res,
+      cacheKey,
+      async () => {
+        const swaps = await req.network.sdex.querySwapsByUser(user, limit, cursor || 0);
```

Comment:
swaps should be retrieved from swaps_v2 table which already has this data, not from subgraph